rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -0,0 +1,295 @@
+import logging
+from pathlib import Path
+from typing import Text, Any, List, Dict, Tuple, TYPE_CHECKING, Union
+
+from ruamel.yaml import YAMLError
+
+import rasa.utils.io as io_utils
+from rasa.constants import DOCS_URL_TRAINING_DATA_NLU
+from rasa.data import YAML_FILE_EXTENSIONS
+from rasa.nlu.training_data.formats.readerwriter import TrainingDataReader
+from rasa.utils.common import raise_warning
+
+if TYPE_CHECKING:
+    from rasa.nlu.training_data import TrainingData
+
+
+logger = logging.getLogger(__name__)
+
+KEY_NLU = "nlu"
+KEY_INTENT = "intent"
+KEY_INTENT_EXAMPLES = "examples"
+KEY_INTENT_TEXT = "text"
+KEY_SYNONYM = "synonym"
+KEY_SYNONYM_EXAMPLES = "examples"
+KEY_REGEX = "regex"
+KEY_REGEX_EXAMPLES = "examples"
+KEY_LOOKUP = "lookup"
+KEY_LOOKUP_EXAMPLES = "examples"
+
+
+class RasaYAMLReader(TrainingDataReader):
+    """Reads YAML training data and creates a TrainingData object."""
+
+    def _initialize(self) -> None:
+        self.training_examples = []
+        self.entity_synonyms = {}
+        self.regex_features = []
+        self.lookup_tables = []
+
+    def reads(self, string: Text, **kwargs: Any) -> "TrainingData":
+        """Reads TrainingData in YAML format from a string
+
+        Args:
+            string: String with YAML training data.
+            **kwargs: key-value args.
+
+        Returns:
+            New `TrainingData` object with parsed training data.
+        """
+        from rasa.nlu.training_data import TrainingData
+
+        self._initialize()
+
+        yaml_content = io_utils.read_yaml(string)
+
+        for key, value in yaml_content.items():  # pytype: disable=attribute-error
```

Comment:
Do we need to check that it's actually a dict, or does this happen automatically before `reads()` is called?

Reply:
That doesn't happen automatically, but I plan to ultimately move this responsibility to the schema verification.